lib/semian/lru_cache.rb

```diff
@@ -0,0 +1,50 @@
+class LRUCache
+  attr_accessor :max_items, :table
+  def initialize(max_items)
+    @max_items = max_items
+    @table = {}
+  end
+
+  def set(key, value)
+    @table.delete(key)
+    @table[key] = value
+    @table.shift if @table.length > @max_items
+    value
+  end
+
+  def get(key)
+    found = true
+    value = @table.delete(key){ found = false }
```

Comment:
I don't see the point of this fancy `found` stuff.

`{}.delete("key") == nil`

So you can effectively do `found = @table.delete(key)` no?